configman/config_manager.py

```diff
@@ -675,7 +686,9 @@ def _overlay_expand(self):
                 if key in all_reference_values:
                     # make sure that this value gets propagated to keys
                     # even if the keys have already been overlaid
-                    known_keys -= set(all_reference_values[key])
+                    finished_keys -= set(
+                        all_reference_values[key]
+                    )
```

Comment:
This is why you're SO RIGHT! I **READ** `FOR key in all_reference_values:` not `IF key in all_reference_values:`